tests/test_api.py

```diff
@@ -0,0 +1,105 @@
+import base64
+
+import pytest
+import swapper
+
+from fcm_django.models import DeviceType
+
+FCMDevice = swapper.load_model("fcm_django", "fcmdevice")
+
+
+@pytest.fixture
+def username() -> str:
+    return "someone"
+
+
+@pytest.fixture
+def password() -> str:
+    return "something"
+
+
+@pytest.fixture
+def user(django_user_model, username: str, password: str):
+    return django_user_model.objects.create_user(username=username, password=password)
+
+
+@pytest.mark.django_db
+def test_tastypie_endpoint_add_device(
+    client, user, username, password, registration_id
+):
+    devices_qty = FCMDevice.objects.count()
+
+    response = client.post(
+        "/tastypie/v1/device/apns/",
+        {
+            "registration_id": registration_id,
+            "type": "web",
+        },
+        content_type="application/json",
+        HTTP_AUTHORIZATION="Basic "
+        + base64.b64encode(f"{username}:{password}".encode()).decode("utf-8"),
+    )
+
+    assert response.status_code == 201
+    assert FCMDevice.objects.count() == devices_qty + 1
+    device = FCMDevice.objects.get(registration_id=registration_id)
+    assert device.type == DeviceType.WEB
+    assert device.registration_id == registration_id
+
+
+@pytest.mark.django_db
+def test_drf_endpoint_add_device(client, registration_id):
```

Comment:
It would be beneficial to have a separate file for DRF and a separate file for tastypie tests

Reply:
there's not much to separate just a single test for tastypie

Reply:
> Honestly, there is too much stuff happening in this PR.

Yep, that's true.

> Tests for api endpoints should be handled in a separate PR.

Integration of swapper could fail functionality for an API. There were no tests for an API so I added it.

> And I agree with @xtrinch that separate files are better cause we are testing completely unrelated concepts.

I split it into two files.